sycl/source/detail/device_impl.cpp

```diff
@@ -855,6 +855,92 @@ bool device_impl::extOneapiCanCompile(
   }
 }
 
+sycl::ext::oneapi::experimental::forward_progress_guarantee
+device_impl::getHostProgressGuarantee(
+    ext::oneapi::experimental::execution_scope threadScope,
+    ext::oneapi::experimental::execution_scope coordinationScope) {
+  return sycl::ext::oneapi::experimental::forward_progress_guarantee::
+      weakly_parallel;
+}
+
+// Returns the strongest progress guarantee that can be provided for threads
+// created at threadScope from the coordination scope given by
+// coordinationScope.
+sycl::ext::oneapi::experimental::forward_progress_guarantee
+device_impl::getProgressGuarantee(
+    ext::oneapi::experimental::execution_scope threadScope,
+    ext::oneapi::experimental::execution_scope coordinationScope) const {
+  if (MIsHostDevice)
```

Comment:
Host device was part of SYCL 1.2 but was removed in SYCL2020. 
We plan to remove any signs of it in RT source code soon.  Would be glad to not introduce new code for it.

tagging @aelovikov-intel for awareness.